storage/storage.go

```diff
@@ -939,13 +942,29 @@ func prepareInsertRecommendationsStatement(
 	return
 }
 
+func getImpactedSinceSQL(ruleFqdn string,
+	errorKey ctypes.ErrorKey,
+	impactedSinceMap map[string]ctypes.Timestamp,
+) (impactedSinceSQL sql.NullTime) {
+
+	impactedSinceSQL = sql.NullTime{}
+	var err error
+	if impactedSince, ok := impactedSinceMap[ruleFqdn+string(errorKey)]; ok {
+		impactedSinceSQL.Time, err = time.Parse(time.RFC3339, string(impactedSince))
```

Comment:
I'd follow the "happy" path there

1) test item existence in math + preliminary return
2) parse time + preliminary return